js/gdm/loginDialog.js

```diff
@@ -688,12 +711,105 @@ const LoginDialog = new Lang.Class({
                                    this._onUserListActivated(item);
                                }));
 
+        this._customerSupportEmail = null;
+        this._customerSupportPhoneNumber = null;
    },
 
     _showPasswordHint: function() {
         this._passwordHintLabel.set_text(this._user.get_password_hint());
         this._passwordHintLabel.show();
         this._passwordHintButton.hide();
+        this._maybeShowPasswordResetButton();
+    },
+
+    _readCustomerSupportData: function() {
+        try {
+            let keyFile = new GLib.KeyFile();
+            keyFile.load_from_file(Config.PKGDATADIR + '/vendor-customer-support.ini', GLib.KeyFileFlags.NONE);
+            this._customerSupportEmail = keyFile.get_locale_string('Customer Support', 'Email', null);
+            this._customerSupportPhoneNumber = keyFile.get_locale_string('Customer Support', 'Phone', null);
+        } catch (e) {
+            logError(e, 'Failed to read customer support data');
+            return false;
+        }
+
+        return this._customerSupportEmail && this._customerSupportPhoneNumber;
+    },
+
+    _maybeShowPasswordResetButton: function() {
+        if (!this._customerSupportPhoneNumber || !this._customerSupportEmail) {
+            if (!this._readCustomerSupportData())
+                return;
+        }
+
+        // There's got to be a better way to get our pid...?
+        let credentials = new Gio.Credentials();
+        let pid = credentials.get_unix_pid();
+
+        // accountsservice provides no async API, and unconditionally informs
+        // polkit that interactive authorization is permissible. If interactive
+        // authorization is attempted on the login screen during the call to
+        // set_password_mode, it will hang forever. Ensure the password reset
+        // button is hidden in this case. Besides, it's stupid to prompt for a
+        // password in order to perform password reset.
+        Polkit.Permission.new('org.freedesktop.accounts.user-administration',
+                              Polkit.UnixProcess.new_for_owner(pid, 0, -1),
+                              null,
+                              Lang.bind(this, function (obj, result) {
+                                  try {
+                                      let permission = Polkit.Permission.new_finish(result);
+                                      if (permission.get_allowed())
+                                          this._passwordResetButton.show();
+                                  } catch(e) {
+                                      logError(e, 'Failed to determine if password reset is allowed');
```

Comment:
I'm not sure about that... then I'd want to rename the function to _maybeShowOrHidePasswordResetButton!